### PR TITLE
Hook-up MigrationHandler API

### LINKED ISSUE #1526: Create API endpoints in the server to export and import Airbyte Configuration files for upgrades
## Tell us about the problem you're trying to solve
As part of #1402, we need to expose API endpoints to trigger export and import methods from the UI.

## Describe the solution you’d like
- An API endpoint in the airbyte Server to trigger exports 
- An API endpoint in the airbyte Server to trigger imports

The export/import archive files produced by airbyte-migration may contain multiple files but they should be compressed into a single archive file including the VERSION tag in a file.

This tarball would then be downloadable/uploadable through the UI so that migration scripts can be applied on top of it externally with a CLI tool.

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
Closes #1526

### CODE REVIEW

File: airbyte-api/src/main/openapi/config.yaml [hunk 1]
```diff
@@ -995,6 +997,34 @@ paths:
...
+        - migration
+      summary: Import Airbyte Configuration and Data
+      operationId: importAirbyte
+      requestBody:
+        content:
+          application/x-gzip:
+            schema:
+              $ref: "#/components/schemas/MigrationArchive"
+        required: true
+      responses:
```
Comment (michel-tricot) [inline airbyte-api/src/main/openapi/config.yaml:1025]: can we make it return an json response object? can be an empty object for now but this way we have an homogeneous API and we might need in the future to return some status

**Review by michel-tricot — APPROVED** [review]

File: airbyte-api/src/main/openapi/config.yaml [hunk 2]
```diff
@@ -52,6 +52,8 @@ tags:
     description: Connection between sources and destinations.
   - name: health
     description: Healthchecks
+  - name: migration
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:55]: i would suggest we do not use the word "migrate" here. Migrate is describing what we will be using the endpoints _for_ not what the endpoints actually return  _is_. 

In the API I would suggest we use a namespace such as: `data`, `archive` or something like that.

Comment (cgardens) [reply airbyte-api/src/main/openapi/config.yaml:55]: Was looking for what other companies do. Cloudera Manager call [this namespace](https://docs.cloudera.com/cloudera-manager/7.0.1/concepts/topics/cm-api-backup-restore-configuration.html) `deployment` which i like well enough.

File: airbyte-api/src/main/openapi/config.yaml [hunk 3]
```diff
@@ -995,6 +997,38 @@ paths:
                 $ref: "#/components/schemas/HealthCheckRead"
         "404":
           description: Debug info not found
+  /v1/migration/export:
+    post:
+      tags:
+        - migration
+      summary: Export Airbyte Configuration and Data
+      operationId: exportAirbyte
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:1005]: we should come up with a name for what the bundled data is called. I think you had previously used the word `archive`. I thought that was catchy? Should we use that? 

If so then `exportAirbyte` => `exportArchive`, `importAirbyte` => `importArchive`

File: airbyte-api/src/main/openapi/config.yaml [hunk 4]
```diff
@@ -995,6 +997,38 @@ paths:
...
+      responses:
+        "200":
+          description: Successful operation
+          content:
+            application/x-gzip:
+              schema:
+                $ref: "#/components/schemas/MigrationArchive"
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:1012]: just `Archive`? or if necessary (if archive is a reserved word), maybe `AirbyteArchive`?

**Review by cgardens — APPROVED** [review]
looks good. some suggestions on names, but otherwise good.

File: airbyte-server/src/main/java/io/airbyte/server/apis/ConfigurationApi.java [hunk 5]
```diff
@@ -385,6 +391,18 @@ public WbConnectionRead webBackendGetConnection(@Valid ConnectionIdRequestBody c
     return execute(() -> webBackendConnectionsHandler.webBackendGetConnection(connectionIdRequestBody));
   }
 
+  // MIGRATION
```
Comment (michel-tricot) [inline airbyte-server/src/main/java/io/airbyte/server/apis/ConfigurationApi.java:394]: We should really remove any mention of "migration" it it just a config import/export

**Review by michel-tricot — APPROVED** [review]

**Pull request merged**